lib/iex/lib/iex/introspection.ex

```diff
@@ -410,15 +410,19 @@ defmodule IEx.Introspection do
     end
   end
 
-  defp print_callback_doc(name, kind, doc, key, callbacks, printer) do
-    {_, [spec | _]} = List.keyfind(callbacks, key, 0)
+  defp print_callback_docs(name, kind, doc, key, callbacks, printer) do
+    {_, specs} = List.keyfind(callbacks, key, 0)
 
-    definition =
-      Typespec.spec_to_ast(name, spec)
-      |> Macro.prewalk(&drop_macro_env/1)
-      |> Macro.to_string
+    specs
+    |> Enum.reverse()
```

Comment:
The multiple callback clauses come the reverse of their definition order, so I reverse it to show.

Reply:
Oh, good catch. I have pushed a later commit that ensures we write the original order to the beam file.